tests/integration/offer/test_offer.py

```diff
@@ -0,0 +1,15 @@
+from django.test import TestCase
+
+from oscar.test.factories import create_product
+from oscar.test.factories.offer import ConditionalOfferFactory
+
+
+class TestOffer(TestCase):
+
+    def test_non_public_product_not_in_offer(self):
+        offer = ConditionalOfferFactory()
+        product = create_product(is_public=False)
+        offer.condition.range.add_product(product)
+        self.assertFalse(product in offer.products())
+        self.assertFalse(
```

Comment:
This is not sufficient unfortunately, you need to test the RangeDetail view. this is just executing the code you added to the view.